simapp/v2/simdv2/cmd/commands.go

```diff
@@ -92,13 +92,13 @@ func genesisCommand[T transaction.Tx](
 		jailAllowedAddrs []string,
 		viper *viper.Viper,
 		modulesToExport []string,
-	) (servertypes.ExportedApp, error),
+	) (serverv2.ExportedApp, error),
 	cmds ...*cobra.Command,
 ) *cobra.Command {
-	compatAppExporter := func(logger log.Logger, db dbm.DB, traceWriter io.Writer, height int64, forZeroHeight bool, jailAllowedAddrs []string, appOpts servertypes.AppOptions, modulesToExport []string) (servertypes.ExportedApp, error) {
+	compatAppExporter := func(logger log.Logger, db dbm.DB, traceWriter io.Writer, height int64, forZeroHeight bool, jailAllowedAddrs []string, appOpts servertypes.AppOptions, modulesToExport []string) (serverv2.ExportedApp, error) {
```

Comment:
this was here for compatibility, we should modify this to strictly what we need.
Additionally we don't need AppOptions anymore and can just use viper.